Fix `System::kernel_long_version`

GuillaumeGomez opened the pull request:
Follow-up of [#1479](https://github.com/GuillaumeGomez/sysinfo/pull/1479).
Part of #1417.

cc @dtolnay

dtolnay commented on src/common/system.rs:
```diff
@@ -847,6 +819,35 @@ impl System {
 
+    /// Provides kernel version following this string format:
+    ///
+    /// "<code>[distribution_id] OS Build [kernel_version]</code>"
```
This explanation is no longer accurate.

It would be helpful to include a table of example platforms like https://docs.rs/sysinfo/0.33.1/sysinfo/struct.System.html#method.kernel_version

dtolnay reviewed (commented):
Thanks! This looks good except the doc comment.

GuillaumeGomez commented:
I changed the output  bit to include the OS version and the kernel version in parens. Does it still look like what you want?

dtolnay commented:
For my use case and https://github.com/sharkdp/bugreport, adding the OS would defeat the purpose. The point was to have the same key for 2 different OS using the same kernel. (And different key for 2 systems using different kernels.)

dtolnay commented on src/common/system.rs:
```diff
@@ -847,6 +819,43 @@ impl System {
+    /// | Windows | Windows 10 (OS Build 20348.2227) |
+    /// | Linux | Linux 41 (kernel version 6.12.13-200.fc41.x86_64) |
+    /// | Android | Android 6.12.13-200 (kernel version 612.13-200) |
+    /// | MacOS | Darwin 12.6.3 (kernel version 21.6.0) |
```
It is a version number of macOS, not a version of Darwin. Rendering that information as "Darwin 12.6.3" is not correct.
https://en.wikipedia.org/wiki/MacOS_Monterey#Release_history

GuillaumeGomez replied:
Ah yes, like you mentioned above, I'm planning on only keeping the kernel version. So I'll update the docs.

GuillaumeGomez commented:
Updated the docs and removed the system version. Not sure why I put it there in the first place, monday... Anyway, hopefully should be what you expect now,

dtolnay commented on src/common/system.rs:
```diff
@@ -847,6 +819,39 @@ impl System {
 
+    /// Provides kernel version following this string format:
+    ///
+    /// | Windows | Windows OS Build 20348.2227 |
```
This needs a table header or the markdown won't render as a table.

<img src="https://github.com/user-attachments/assets/2ada445a-aa51-4b85-aa13-c371505c0052">

```suggestion
    /// | example platform | value of `System::kernel_long_version()` |
    /// |---|---|
    /// | Windows | Windows OS Build 20348.2227 |
```

Or write the table using html table tags.

GuillaumeGomez replied:
I was sure it could be skipped in markdown. Thanks for the heads up!

GuillaumeGomez commented:
Thanks a lot for your help @dtolnay !
